client/src/pages/LyricsView/LyricsDisplay.tsx

```diff
@@ -52,6 +64,57 @@ const LyricsDisplay: React.FunctionComponent<IProps> = ({
     }
   }, [searchShown]);
 
+  //accounting for rewinding the song
+  useEffect(() => {
+    if (progressMs === undefined) {
+      return; // song switched
+    }
+    if (progressMs < naturalSongProgress) {
+      const clonedLyricsArray = syncedLyricsArray === undefined ? [] : [...syncedLyricsArray];
+      setNaturalSongProgress(0);
+      setSyncedLyrics({ array: clonedLyricsArray, before: "", highlighted: "", after: lyrics });
+    }
+  }, [progressMs]);
+
+  //making sure highlighted text is always visible
+  useEffect(() => {
+    const element = document.getElementById("highlighted-text");
```

Comment:
Rather than `document.getElementById`, we should use a ref in React

Reply:
Thanks! it's been 5 years since I last developed in react (and still I was merely a manger helping!) learning a lot with you here!